python/semantic_kernel/connectors/memory/azure_cog_search/acs_memory_store.py

```diff
@@ -0,0 +1,489 @@
+        """
+        record._key = record._id
+
+        collection_info = self.does_collection_exist_async(
+            collection_name=collection_name
+        )
+
+        if not collection_info:
+            raise Exception(f"Collection '{collection_name}' does not exist")
+
+        if record._id:
+            id = record._id
+        else:
+            id = uuid.uuid4()
+
+        acs_doc = {
+            "collection_name": collection_name,
+            "timestamp": datetime.utcnow(),
+            "vector_id": id,
+            "payload": str(json.load(record._payload)),
+            "vector": record._embedding.tolist(),
+        }
+
+        result = self._cogsearch_client.upload_documents(documents=[acs_doc])
+
+        if result[0].succeeded:
+            return id
+        else:
+            return ""
+
+    async def upsert_batch_async(
+        self, collection_name: str, records: List[MemoryRecord]
+    ) -> List[str]:
+        """Upserts a batch of records.
+
+        Arguments:
+            collection_name {str} -- The name of the collection to upsert the records into.
+            records {List[MemoryRecord]} -- The records to upsert.
+
+        Returns:
+            List[str] -- The unique database keys of the records.
+        """
+
+        collection_result = self.does_collection_exist_async(
+            collection_name=collection_name
+        )
+
+        if not collection_result:
+            raise Exception(f"Collection '{collection_name}' does not exist")
+
+        acs_documents = []
+        acs_ids = []
+
+        for record in records:
+            if record._id:
+                id = record._id
+            else:
+                id = uuid.uuid4()
+
+            acs_ids.append(id)
+
+            acs_doc = {
+                "collection_name": collection_name,
+                "timestamp": datetime.utcnow(),
+                "vector_id": str(id),
+                "payload": str(json.load(record._payload)),
+                "vector": record._embedding.tolist(),
+            }
+
+            acs_documents.append(acs_doc)
+
+        result = self._cogsearch_client.upload_documents(documents=[acs_doc])
+
+        if result[0].succeeded:
+            return acs_ids
+        else:
+            return None
+
+    async def get_async(
+        self, collection_name: str, key: str, with_embedding: bool = True
+    ) -> MemoryRecord:
+        """Gets a record.
+
+        Arguments:
+            collection_name {str} -- The name of the collection to get the record from.
+            key {str} -- The unique database key of the record.
+            with_embedding {bool} -- Whether to include the embedding in the result. (default: {False})
+
+        Returns:
+            MemoryRecord -- The record.
+        """
+
+        search_id = key
+
+        results = self._cogsearch_client.search(
+            search_text="",
+            vector=Vector(value=search_id, k=3, fields="collection_id"),
```

Comment:
why is k fixed to 3? shouldn't this be a parameter passed in through semantic kernel?

Reply:
Thanks @mattmsft Making all the changes we discussed. You rock.

Reply:
Changed to parameter passing in limit: https://github.com/microsoft/semantic-kernel/pull/1103/commits/ca595c56df60c1c0fa3c3adba706dc8e358886ea